tests/4.5/target_teams_distribute/test_target_teams_distribute_firstprivate.F90

```diff
@@ -98,16 +98,26 @@ INTEGER FUNCTION test_firstprivate_first()
     END DO
 
     !$omp target data map(from: d(1:N)) map(to: a(1:N), b(1:N), c(1:N))
-    !$omp target teams distribute firstprivate(privatized_array, privatized) &
+    !$omp target teams distribute private(idx) firstprivate(privatized_array, privatized) &
     !$omp map(alloc: a(1:N), b(1:N), c(1:N), d(1:N)) num_teams(10)
     DO x = 1, N
        num_teams(x) = omp_get_num_teams()
-       d(x) = a(x) + b(x) + c(x) + privatized_array(MOD(x, 10)) + privatized
+       IF (MOD(x, 10).eq.0) THEN
+         idx = 1;
+       ELSE 
+         idx = MOD(x, 10)
+       END IF
+       d(x) = a(x) + b(x) + c(x) + privatized_array(idx) + privatized
```

Comment:
I think it should be just 
```FORTRAN
       d(x) = a(x) + b(x) + c(x) + privatized_array(MOD(x, 10)+1) + privatized	
```

Since arrays are not 0 indexed, but 1 indexed.

Reply:
Yup, this will work.